addons/account/static/src/js/tax_group.js

```diff
@@ -75,97 +144,93 @@ odoo.define('account.tax_group', function (require) {
          */
-        _render: function () {
-            var self = this;
-            // Display the pencil and allow the event to click and edit only on purchase that are not posted and in edit mode.
-            // since the field is readonly its mode will always be readonly. Therefore we have to use a trick by checking the 
-            // formRenderer (the parent) and check if it is in edit in order to know the correct mode.
-            var displayEditWidget = self._isPurchaseDocument() && this.record.data.state === 'draft' && this.getParent().mode === 'edit';
-            this.$el.html($(QWeb.render('AccountTaxGroupTemplate', {
-                lines: self.value,
-                displayEditWidget: displayEditWidget,
-            })));
-        },
-
-        //--------------------------------------------------------------------------
-        // Handler
-        //--------------------------------------------------------------------------
+        _displayEditWidget() {
+            return this._isPurchaseDocument() && this.record.data.state === 'draft' && this._getParentWidget().mode === 'edit';
+        }
 
         /**
-         * This method is called when the user is in edit mode and 
-         * leaves the <input> field. Then, we execute the code that 
-         * modifies the information.
-         * 
-         * @param {event} ev 
+         * This method formats a list of list of tax groups into a list of objects.
+         * This job is done to be more understandable in the code.
          */
-        _onBlur: function (ev) {
-            ev.preventDefault();
-            var $input = $(ev.target);
-            var newValue = $input.val();
-            var currency = session.get_currency(this.record.data.currency_id.data.id);
-            try {
-                newValue = fieldUtils.parse.float(newValue);    // Need a float for format the value.             
-                newValue = fieldUtils.format.float(newValue, null, {digits: currency.digits}); // return a string rounded to currency precision
-                newValue = fieldUtils.parse.float(newValue); // convert back to Float to compare with oldValue to know if value has changed
-            } catch (err) {
-                $input.addClass('o_field_invalid');
-                return;
-            }
-            var oldValue = $input.data('originalValue');
-            if (newValue === oldValue || newValue === 0) {
-                return this._render();
-            }
-            var taxGroupId = $input.parents('.oe_tax_group_editable').data('taxGroupId');
-            this._changeTaxValueByTaxGroup(taxGroupId, oldValue-newValue);
-        },
+        _formatTaxGroups() {
+            let formatedTaxGroups = [];
+            this.value.forEach(element => {
+                formatedTaxGroups.push({
+                    id: element[6],
+                    name: element[0],
+                    taxValue: element[1],
+                    taxDisplay: element[3],
```

Comment:
Ouch. Is it  not possible to provide the keys in `_compute_invoice_taxes_by_group` instead of fetching by id?

Reply:
In the first version of the widget, we had already considered changing the `amount_by_group` format but we preferred to avoid it. Maybe talk with @qdp-odoo about it and make another task for this change?